core/src/main/scala/kafka/coordinator/transaction/TransactionStateManager.scala

```diff
@@ -310,8 +302,9 @@ class TransactionStateManager(brokerId: Int,
 
   // TODO: check broker message format and error if < V2
   def appendTransactionToLog(transactionalId: String,
+                             coordinatorEpoch: Int,
```

Comment:
I have been thinking about it, the main reason that I keep it separately is that it can be called in two paths: handler thread call it directly and the callback from writeMarkerSender call it separately. In the latter case the two values are separate so I ended up keep them as is.